### ISSUE TITLE
custom loss function

### ISSUE THREAD

**Issue by slaweks17** [Microsoft/CNTK#715]
Hi,
Is there any way to specify own loss function in CNTK? (e.g. add penalty for weights or companion losses per layer)

Regards,
Slawek

**Comment by frankseide** [comment]
Any loss function that can be expressed as a BrainScript or NDL expression can be used. Just declare that value in the criterionNodes variable. For example, this is how the cross-entropy loss would be written in this way (this form is used in the [sequence-to-sequence](https://github.com/Microsoft/CNTK/tree/master/Examples/SequenceToSequence/CMUDict/Config) example):

    ce   = ReduceLogSum (z) - TransposeTimes (labels, z)
    criterionNodes = (ce)

where z is the input to the Softmax function.

**Comment by slaweks17** [comment]
Great! Thank you.


________________________________
Od: Frank Seide <notifications@github.com>
Wysłane: 25 lipca 2016 16:01
Do: Microsoft/CNTK
DW: slaweks17; Author
Temat: Re: [Microsoft/CNTK] custom loss function (#715)


Any loss function that can be expressed as a BrainScript or NDL expression can be used. Just declare that value in the criterionNodes variable. For example, this is how the cross-entropy loss would be written in this way (this form is used in the sequence-to-sequence<https://github.com/Microsoft/CNTK/tree/master/Examples/SequenceToSequence/CMUDict/Config> example):

ce   = ReduceLogSum (z) - TransposeTimes (labels, z)
criterionNodes = (ce)


where z is the input to the Softmax function.

-

**Comment by frankseide** [comment]
Please do give it a shot. If it works, would you mind closing the issue? If it does not work, please let's continue this thread to get it to work for you.
